### ISSUE TITLE
record_rule_instantiation_callstack disappeared without any note

### ISSUE THREAD

**Issue by limdor** [bazelbuild/bazel#12721]
### Description of the problem / feature request:

record_rule_instantiation_callstack is available in 3.6.0 but not available in 3.7.0

### Feature requests: what underlying problem are you trying to solve with this feature?

Add a note in the release notes saying that is not available anymore and explain what is the alternative or if this is the default behavior

### What's the output of `bazel info release`?

3.7.0

###  Have you found anything relevant by searching the web?

This is the commit that removes the option and says that it was already a no opts option.
https://github.com/bazelbuild/bazel/commit/35f504780e17d24e8d19617becf5bd4a8fe028d3
To me it is not clear if the record of rule instantiation is already the default or there is another way to do it.
Having in the release notes of 3.7.0 would help the user.

Checking in the release notes of 3.6.0 is indeed encouraging to use the option in a very specific case:
https://github.com/bazelbuild/bazel/blob/master/CHANGELOG.md#release-360-2020-10-06

**Comment by benjaminp** [comment]
It's the default: 5b236638e115ed1ae6c586722ac5c126bffcf159

**Comment by jin** [comment]
cc @murali42 @alandonovan 

Could you please share some context about the flag removal and/or update the 3.7.0 release notes about this?

**Comment by murali42** [comment]
TLDR: --record_rule_instantiation_callstack is the default (we got enough memory savings from elsewhere we could enable it), hence was removed.

Few releases ago that flag would control whether the callstack was recorded and whether query output would include it. 
Now  the callstack is always recorded and `query --proto:instantiation_stack` controls whether it is included in the output.

**Comment by jin** [comment]
Thanks @murali42! Looks like the flag should have been removed as part of a major version change to 4.0 since it's an incompatible change. It would be helpful to mention flag removal reasons in the release notes.

Closing.
